HS_PRMS SKU in azurerm_mssql_elasticpool gives the error This is not a valid sku name

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.6.6

### AzureRM Provider Version

3.87.0

### Affected Resource(s)/Data Source(s)

azurerm_mssql_elasticpool

### Terraform Configuration Files

```hcl
resource "azurerm_mssql_elasticpool" "main" {
  depends_on          = [azurerm_mssql_server.main]
  name                = "ep${var.company_name}${var.envtype}"
  resource_group_name = var.rgname
  location            = var.region
  server_name         = azurerm_mssql_server.main.name
  max_size_gb         = var.ep_size
  license_type        = var.ep_license
  sku {
    name     = var.ep_name   
    tier     = var.ep_edition  
    family   = var.ep_compute_gen 
    capacity = var.ep_cores       
  }
  per_database_settings {
    min_capacity = "0"
    max_capacity = var.ep_cores
  }
  tags = {
    EnvType       = var.envtype
    SiteName      = "${var.cloudsitename}.${var.domain}"
    SuppressUntil = ""
  }
  lifecycle {
    ignore_changes = [tags]
  }
}
```


### Debug Output/Panic Output

```shell
╵
╷
│ Error: expected sku.0.name to be one of ["BasicPool" "StandardPool" "PremiumPool" "GP_Gen4" "GP_Gen5" "GP_Fsv2" "GP_DC" "BC_Gen4" "BC_Gen5" "BC_DC" "HS_Gen5"], got HS_PRMS
│ 
│   with module.SQL.azurerm_mssql_elasticpool.main,
│   on modules/SQL/main.tf line 55, in resource "azurerm_mssql_elasticpool" "main":
│   55:     name     = var.ep_name        
│ 
╵
╷
│ Error: Cannot apply incomplete plan
```


### Expected Behaviour

Resource should have been deployed with sku HS_PRMS

### Actual Behaviour

##[error]custom_ep_sku variable is not properly cased, or not one of the following: GP_Gen4 GP_Gen5 GP_Fsv2 GP_DC BC_Gen4 BC_Gen5 BC_DC HS_Gen5. Value provided was: HS_PRMS

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/mssql/mssql_elasticpool_resource.go
+++ b/internal/services/mssql/mssql_elasticpool_resource.go
@@ -90,6 +90,7 @@ func resourceMsSqlElasticPool() *pluginsdk.Resource {
 								"BC_Gen5",
 								"BC_DC",
 								"HS_Gen5",
+								"HS_PRMS",
 							}, false),
 						},
 

--- a/website/docs/r/mssql_elasticpool.html.markdown
+++ b/website/docs/r/mssql_elasticpool.html.markdown
@@ -91,7 +91,7 @@ The following arguments are supported:
 
 The `sku` block supports the following:
 
-* `name` - (Required) Specifies the SKU Name for this Elasticpool. The name of the SKU, will be either `vCore` based or `DTU` based. Possible `DTU` based values are `BasicPool`, `StandardPool`, `PremiumPool` while possible `vCore` based values are `GP_Gen4`, `GP_Gen5`, `GP_Fsv2`, `GP_DC`, `BC_Gen4`, `BC_Gen5`, `BC_DC`, or `HS_Gen5`.
+* `name` - (Required) Specifies the SKU Name for this Elasticpool. The name of the SKU, will be either `vCore` based or `DTU` based. Possible `DTU` based values are `BasicPool`, `StandardPool`, `PremiumPool` while possible `vCore` based values are `GP_Gen4`, `GP_Gen5`, `GP_Fsv2`, `GP_DC`, `BC_Gen4`, `BC_Gen5`, `BC_DC`, `HS_PRMS`, or `HS_Gen5`.
 
 * `capacity` - (Required) The scale up/out capacity, representing server's compute units. For more information see the documentation for your Elasticpool configuration: [vCore-based](https://docs.microsoft.com/azure/sql-database/sql-database-vcore-resource-limits-elastic-pools) or [DTU-based](https://docs.microsoft.com/azure/sql-database/sql-database-dtu-resource-limits-elastic-pools).
 

